src/rsz/src/Resizer.cc

```diff
@@ -501,14 +502,34 @@ void Resizer::findBuffers()
 {
   if (buffer_cells_.empty()) {
     LibertyLibraryIterator* lib_iter = network_->libertyLibraryIterator();
+    std::unique_ptr<sta::PatternMatch> clkbuf_pattern;
+
+    if (exclude_clock_buffers_) {
+      clkbuf_pattern
+          = std::make_unique<sta::PatternMatch>(".*CLKBUF.*",
+                                                /* is_regexp */ true,
+                                                /* nocase */ true,
+                                                /* Tcl_interp* */ nullptr);
+    }
```

Comment:
I really feel strongly that this should be refactored into it's own method that returns something that can be queried for function.

Something like 

```c++
CellFunction GetCellFunction(sta::LibertyCell* | dbMaster*);

CellFunction function = GetCellFunction(buffer);

if (function->isClockBuffer(buffers)) {
 ...
}
```

I think otherwise we risk spreading around this kind of adhoc code everywhere and making a mess.

Reply:
Honestly, I'd prefer if we didn't have this regex thing here at all, but since it seems like the consensus is the project would like it add it, I strongly thing we should do the above to limit the blast radius.

Reply:
@QuantamHD Within this logic would we want to keep this code in the RSZ or move it to dbSTA? It seems like it would more sense for it to stay in the latter. I don't mind making the change and it does looks cleaner to me. @maliberty Any thoughts?

Reply:
Making it maximally accessible in dbSta seems like a good change since it would reduce friction to using it.

Reply:
I view the regex as temporary means of determining what is a clock buffer.  We could take it from the user or some internal analysis in the future.  Encapsulating it seems fine.

Reply:
"Nothing more permanent than a temporary fix."

Reply:
Will there be an issue opened to track the replacement of the regex? I would worry that without some kind of tracking for this, this regex will just linger as an "okay" solution so no need to go back and fix it.

Reply:
@AcKoucher make an issue please

Reply:
Opened https://github.com/The-OpenROAD-Project/OpenROAD/issues/5753